src/environment/environment.js

```diff
@@ -0,0 +1,127 @@
+/** @flow */
+import os from 'os';
+import { v4 } from 'uuid';
+import fs from 'fs-extra';
+import path from 'path';
+import npmInstall from '../utils/npm';
+import { loadScope } from '../scope';
+import { flattenDependencies } from '../scope/flatten-dependencies';
+import { BitId } from '../bit-id';
+import { Component } from '../consumer/component/consumer-component'
+import { BITS_DIRNAME } from '../constants';
+import { mergeAll } from 'ramda';
+
+const root = path.join(os.tmpdir(), 'bit');
+const currentPath = process.cwd();
+
+export default class Environment {
+  path: string;
+  component: Component;
+  componentsDependencies: ConsumerComponent[] = [];
+
+  constructor() {
+    this.path = path.join(root, v4());
+    console.log(this.path);
+  }
+
+  init(): Promise<*> {
+    return new Promise((resolve, reject) => {
+      fs.ensureDir(this.path, (err) => {
+        if (err) return reject(err);
+        return resolve();
+      });
+    });
+  }
+
+  importById(rawId: string): Promise<Object> {
+    // loadScope works for local scope and remote scope
+    return loadScope(currentPath).then((scope) => {
+      const bitId = BitId.parse(rawId, scope.name);
+      return scope.get(bitId)
+        .then((component) => {
+          component.component.write(this.path, true, true);
+          this.component = component.component;
+          return component;
+        });
+    });
+  }
+
+  writeToEnvironmentDir(componentDependencies: versionDependencies[]): Promise<Component[]> {
+    const componentsDir = path.join(this.path, BITS_DIRNAME);
+    const components = flattenDependencies(componentDependencies);
+
+    const bitDirForEnvironmentImport = (component: Component) => {
+      return path.join(
+        componentsDir,
+        component.box,
+        component.name,
+        component.scope,
+        component.version.toString(),
+      );
+    };
+
+    return Promise.all(components.map((component) => {
+      const bitPath = bitDirForEnvironmentImport(component);
+      return component.write(bitPath, true);
+    }));
+  }
+
+  importDependencies(): Promise<ConsumerComponent[]> {
+    if (!this.component) return Promise.reject();
```

Comment:
I did some changes, here is how you can use the class now:
```js
const environment = new Environment();
  console.log(environment.getPath());
  return environment
    .init()
    .then(() => console.log(environment.consumer))
    .then(() => environment.importComponent('bit.utils/array/diff'))
    .then(component => environment.installDependencies(component)
      .then(() => environment.installNpmPackages(component))
      .then(() => {
        console.log(environment.getComponentPath(component));
        console.log(environment.getImplPath(component));
        console.log(environment.getMiscFilesPaths(component));
      })
    )
    .then(() => environment.destroy());
```
It's using the Consumer for the most parts, and the consumer is available as a public property.

I preferred to keep the `init()` method for readability (facade pattern). It's easier to understand the class this way as it's a vital part of any use of this class.

Also, as you can see, the methods in this class work with one component. At the beginning, I tried to work with an array of ids, but it got complicated especially with the getImplPath and getMiscPaths. In order to get the path, you'll probably need to generate an array of ids => components after the import and keep it in-sync with the file-system. Otherwise, it'll the responsibility of the one who uses the class to keep track which component is related to what id. 
I believe that if we don't need this ability currently it's better not to add this layer of complexity. (KISS principle).